### ISSUE TITLE
Fix false positives for at-charset and single quotes in string-quotes

### ISSUE THREAD

**Issue by LiHaoTan** [stylelint/stylelint#2788]
When specifying the rule `string-quotes: "single"`, then specifying the encoding through `@charset` single quotes is required as well, e.g. `@charset 'UTF-8';`.

However, this is incorrect as per the specification double quotes are always required:

> Otherwise, check the byte stream. If the first 1024 bytes of the stream begin with the hex sequence
40 63 68 61 72 73 65 74 20 22 XX* 22 3B
where each XX byte is a value between 016 and 2116 inclusive or a value between 2316 and 7F16 inclusive, then get an encoding for the sequence of XX bytes, interpreted as ASCII

https://drafts.csswg.org/css-syntax-3/#determine-the-fallback-encoding

**Comment by ntwb** [comment]
Thanks @LiHaoTan, I've marked this issue up as a bug and help wanted, would you be interested in creating a pull request to fix this?

From the cited source above: (emphasis mine)
> Note: Note that the syntax of an encoding declaration looks like the syntax of an at-rule named `@charset`, but no such rule actually exists, and the rules for how you can write it are much more restrictive than they would normally be for recognizing such a rule. **A number of things you can do in CSS that would produce a valid `@charset` rule (if one existed), such as using multiple spaces, comments, or single quotes, will cause the encoding declaration to not be recognized**. This behavior keeps the encoding declaration as simple as possible, and thus maximizes the likelihood of it being implemented correctly.

**Comment by evilebottnawi** [comment]
@stylelint/core what do your think about this, should we ignore `@charset` or output error if your use `single` for `@charset`?

**Comment by ntwb** [comment]
Generally I think this should throw an error:

```css
@charset 'UTF-8';
```

What about a new rule, `no-invalid-charset-quotes` in the "Possible errors" rule category?

Message `Expected double quotes, @charset encoding declarations must use double quotes`

The `string-quotes` rule would then _ignore_ instances of `@charset`

**Comment by evilebottnawi** [comment]
@ntwb sgtm, I would like to hear one more opinion :smile:

**Comment by jeddy3** [comment]
> The string-quotes rule would then ignore instances of @charset

SGTM

**Comment by jeddy3** [comment]
@LiHaoTan This issue is labelled as "help wanted". If you'd like to see this issue closed, please 
consider [contributing a fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule) i.e. `string-quotes` should ignore instances of `@charset`.

**Comment by jeddy3** [comment]
Well spotted.
